src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -601,32 +895,119 @@ private static PartialCaseDecision RemoveEvaluation(PartialCaseDecision c, Bound
                 Index: c.Index,
                 Syntax: c.Syntax,
                 Decisions: c.Decisions.WhereAsArray(d => !(d is BoundDagEvaluation e2) || e2 != e),
```

Comment:
If the parent method were inlined I suspect this lambda would only end up creating one display class to hold `e` instead of two, which could cut allocations significantly.

Reply:
This captures only `e`.

---
In reply to: [176221928](https://github.com/dotnet/roslyn/pull/24059#discussion_r176221928) [](ancestors = 176221928)